### PR TITLE
modify views page

### PR DESCRIPTION
**Pull request by techEmprez**
### **The following was implemented in this PR**

- Implement the design from the sneak peek wireframes.
    - Styling is nice to have but focus on making sure everything functions first.     

- Use methods that you have created in the [Blog App - processing data in models](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md).

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -73,7 +73,10 @@ GEM
     bootsnap (1.16.0)
       msgpack (~> 1.2)
     builder (3.2.4)
+feature6/views
+    capybara (3.39.1)
```
Comment (CollinsTatang) [inline Gemfile.lock]: Hi @techEmprez 

Kindly permit me to use this Gemfile.lock file to request changes

- [x] Please your User Controller `#index` action should render all Users as seen below.
```ruby
# app/controllers/users_controller.rb
def show
  @user = User.find(params[:id])
end

```
- [x] Please your corresponding `Views` index page should loop through the users and display the following.
```
 # app/views/users/index.html.erb
<% @users.each do |user| %>
# HTML Code Here displaying user.photo, user.name and user.posts_counter
<% end %>
```
- [ ] Clicking on each user takes you to their show page as shown in the [wireframe](https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_user_page.png)
- [ ] Please you should pay more attention on the function as styling is nice to have The index view page should match the given [wireframe](https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_root_users.png)

File: Gemfile.lock [hunk 2]
```diff
@@ -104,10 +107,15 @@ GEM
     jbuilder (2.11.5)
       actionview (>= 5.0.0)
       activesupport (>= 5.0.0)
+ feature6/views
+    loofah (2.21.2)
+      crass (~> 1.0.2)
+      nokogiri (>= 1.12.0)
```
Comment (CollinsTatang) [inline Gemfile.lock]: - [x] Please your User Controller `#show` action should render the user params[:id] as shown below.
```ruby
# app/controllers/users_controller.rb

def show
  @user = User.find(params[:id])
end

```

- [x] Please you `app/views/users/show.html.erb` file should use the `#recent_post` method to render the user's 3 most recent post.
- [x] Please the **See all Post** links opens user's post index as displayed on the [wireframe](https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_user_all_posts.png)

File: Gemfile.lock [hunk 3]
```diff
@@ -129,11 +137,16 @@ GEM
     net-smtp (0.3.3)
       net-protocol
     nio4r (2.5.9)
+feature6/views
+    nokogiri (1.14.4-x64-mingw-ucrt)
+      racc (~> 1.4)
+
```
Comment (CollinsTatang) [inline Gemfile.lock]: - [x] Please use `post.recent_comments` to render 5 most recent comments in the `app/views/posts/index.html.erb` page
```
 <% post.recent_comments.each do |comment| %>
  <p><%= comment.author.name %>: <%= comment.text %></p> 
 <% end %>

```

- [x] Please your post show page displays post with all comments listed as shown on the [wireframe](https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_single_post.png)

**Review by CollinsTatang — CHANGES_REQUESTED** [review]
Hi @techEmprez, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

## Required changes 
It's ok to have bugs in your code. Please, see this code review as a chance for you to learn practical coding skills and learn new tips that can improve your code quality and help you get a better job.
_Check the comments under the review._
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: Gemfile.lock (hunk 1 above)
Comment (afanimartin) [reply Gemfile.lock]: - please make these suggested changes. You haven't yet implemented these changes.

File: Gemfile.lock (hunk 2 above)
Comment (afanimartin) [reply Gemfile.lock]: - Modify the show action as suggested here. It's a good idea to make changes before submitting the PR for review.

**Review by afanimartin — CHANGES_REQUESTED** [review]
Hi @techEmprez, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- Changes made to Post index view as suggested

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

## Find me on slack
### Martin Afani

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: Gemfile.lock (hunk 1 above)
Comment (techEmprez) [reply Gemfile.lock]: All done

**Review by BenMukebo — APPROVED** [review]
### Hi @techEmprez, 👋 

# STATUS: Approved  🎉 

Great work on making the changes requested by a previous reviewer :1st_place_medal:  
 Your project is complete!  There is nothing else to say other than... it's time to merge it
Congratulations! 🎉  🎉 🎉 



### To Highlight :1st_place_medal: 
- Your Pull Request has a proper title and description. ✔️  
- No Linter errors :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- [OPTIONAL] Please improve your  as shown on the [wireframe](https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_single_post.png)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @BenMukebo ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by BenMukebo — APPROVED** [review]
.

**Review by BenMukebo — APPROVED** [review]
.

**Pull request merged**